src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java

```diff
@@ -467,7 +468,9 @@ private static AccessModifierOption getAccessModifierFromModifiersTokenDirectly(
         }
 
         // default access modifier
-        AccessModifierOption accessModifier = AccessModifierOption.PACKAGE;
+        final String scope = ScopeUtil.getScopeFromMods(modifiersToken).toString();
```

Comment:
Please never convert an enum to another via string representation.
This is slow and will be broken after any kind of refactoring.

Secondly: this code is extremely inefficient. `ScopeUtil.getScopeFromMods` returns the declared scope if present. No need for the loop below.

Then, please note that `Scope` is deprecated. See #3511 for the details.

I think that the right way is to make `getAccessModifierFromModifiersTokenDirectly` return `null` if none modifiers present. Then fix `CheckUtil#getAccessModifierFromModifiersToken` to 
```java
    public static AccessModifierOption getAccessModifierFromModifiersToken(DetailAST ast) {
        final DetailAST modsToken = ast.findFirstToken(TokenTypes.MODIFIERS);
        AccessModifierOption accessModifier = getAccessModifierFromModifiersTokenDirectly(modsToken);

        if (accessModifier == null) {
            if (ScopeUtil.isInEnumBlock(ast)) {
                if (ast.getType() == TokenTypes.ENUM_CONSTANT_DEF) {
                    accessModifier = AccessModifierOption.PUBLIC;
                }
                else if (ast.getType() == TokenTypes.CTOR_DEF) {
                    accessModifier = AccessModifierOption.PRIVATE;
                }
                else {
                    accessModifier = AccessModifierOption.PACKAGE;
                }
            }
            else if (ScopeUtil.isInInterfaceOrAnnotationBlock(ast)) {
                accessModifier = AccessModifierOption.PUBLIC;
            }
            else {
                accessModifier = AccessModifierOption.PACKAGE;
            }
        }
        
        return accessModifier;
    }
```
```